samples/coroutines/kotlin/src/main/kotlin/godot/samples/coroutines/Ball.kt

```diff
@@ -4,20 +4,28 @@ import godot.Input
 import godot.Node2D
 import godot.core.Variant
 import godot.core.Vector2
+import org.godotengine.kotlin.annotation.RegisterClass
+import org.godotengine.kotlin.annotation.RegisterFunction
+import org.godotengine.kotlin.annotation.RegisterSignal
 
 
-class Ball : Node2D {
-    constructor() : super()
-    constructor(value: Variant) : super(value)
-
+@RegisterClass("Scripts/")
+class Ball : Node2D() {
 
     var moveSpeed = 2.0
 
+    interface Signal {
+        @RegisterSignal
+        fun move() {}
```

Comment:
The argument is missing for this signal. It does not yield to an error but would if one would implement this interface.